File: Kiro/python/distance_calculator.py
"""Distance calculator for Manhattan distance computations."""

from position import Position


class DistanceCalculator:
    """Calculator for Manhattan distance between grid positions."""
    
    @staticmethod
    def calculate_manhattan_distance(pos1: Position, pos2: Position) -> int:
        """Calculate Manhattan distance between two positions.
        
        Args:
            pos1: First position
            pos2: Second position
            
        Returns:
            Manhattan distance as a non-negative integer
        """
        return abs(pos1.row - pos2.row) + abs(pos1.column - pos2.column)

File: Kiro/go/distance_calculator.go
package gridneighborhoods

// DistanceCalculator provides Manhattan distance calculation
type DistanceCalculator struct{}

// NewDistanceCalculator creates a new distance calculator
func NewDistanceCalculator() *DistanceCalculator {
	return &DistanceCalculator{}
}

// CalculateManhattanDistance computes the Manhattan distance between two positions
func (dc *DistanceCalculator) CalculateManhattanDistance(pos1, pos2 Position) int {
	return pos1.ManhattanDistance(pos2)
}
